client/client.go

```diff
@@ -148,8 +148,15 @@ func (c *Client) init() error {
 			return fmt.Errorf("failed creating state dir: %s", err)
 		}
 
-		c.logger.Printf("[INFO] client: using state directory %v", c.config.StateDir)
+	} else {
```

Comment:
If the user hasn't provided a state directory, perhaps he doesn't want checkpointing features of the Nomad client? Should we be even trying to persist state?